cnxarchive/views.py

```diff
@@ -436,6 +436,256 @@ def format_author(personids, settings):
+                   'cnx-dc-extension.xsd',
+         'namespace': 'http://cnx.rice.edu/cnx_dc/'}
+        ]
+    return_vars['results'] = METADATA_FORMATS
+
+
+def _oaiListSets(return_vars, request):
+    # return error, ListSets currently not supported
+    return_vars['error'] = {'code': 'noSetHierarchy',
+                            'message': 'ListSets not currently supported'}
+
+
+def _oaiListIdentifiers(return_vars, request):
+    _verifyMetadataPrefix(return_vars, request)
+    if 'error' in return_vars.keys():
+        return
+    where, arguments = _setFromUntil(request)
+    statement = """
+                SELECT revised, uuid
+                FROM latest_modules {}
+                ORDER BY revised;
+                """.format(where)
+    settings = request.registry.settings
+    search_results = _databaseDictResults(statement, arguments, settings)
+    if len(search_results) == 0:
+        return_vars['error'] = {'code': 'noRecordsMatch',
+                                'message': 'No matches for the given request'}
+        return
+    return_vars['results'] = search_results
+
+
+def _oaiSearchRecords(return_vars, request):
+    _verifyMetadataPrefix(return_vars, request)
+    if 'error' in return_vars.keys():
+        return
+
+    new_weights = DEFAULT_SEARCH_WEIGHTS
+    if 'weights' in request.GET.keys():
+        # Set the weights to be empty except for the variables set
+        new_weights = {w: 0 for w in DEFAULT_SEARCH_WEIGHTS.keys()}
+        weights = request.GET.get('weights').split(',')
+        for weight in weights:
+            weight_split = weight.split(":", 1)
+            if weight_split[0] in DEFAULT_SEARCH_WEIGHTS.keys():
+                try:
+                    new_weights[weight_split[0]] = int(weight_split[1])
+                except:
+                    pass
+        if all(weight == 0 for weight in new_weights.values()):
+            new_weights = DEFAULT_SEARCH_WEIGHTS
+    query = Query(Query.from_raw_query(request.GET.get('query')))
+    sort = request.GET.get('sorton')
+    if sort in ['pubdate', 'version', 'popularity']:
+        query.sorts = [sort]
+
+    # Build the SQL
+    b_size = request.GET.get('b_size', 10)
+    b_start = request.GET.get('b_start', 0)
+    size = "LIMIT (%(b_size)s) OFFSET (%(b_start)s);"
+    statement, arguments = _build_search(query, new_weights,
+                                         columns=_oaiGetBaseStatement(),
+                                         size=size)
+    arguments['b_size'] = b_size
+    arguments['b_start'] = b_start
+
+    # Execute the SQL.
+    settings = request.registry.settings
+    search_results = _databaseDictResults(statement, arguments, settings,
+                                          is_dict=False)
+    results = []
+    for result in search_results:
+        results.append(result[0])
+    _formatOaiResults(results, return_vars, request)
+
+
+# ################### #
```

Comment:
This is a lot of code for the views file. Perhaps his should live in an oai.py, like search does.